3.evaluate_model/scripts/nbconverted/F1_scores.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### Import Libraries
+
+# In[1]:
+
+
+import pandas as pd
+import pathlib
+from joblib import load
+
+import sys
+sys.path.append("../utils")
+from split_utils import get_features_data
+from train_utils import get_dataset
+from evaluate_utils import model_score
+
+
+# ### Load Necessary Data
+
+# In[2]:
+
+
+# specify results directory
+results_dir = pathlib.Path("evaluations/")
+f1_scores_dir = pathlib.Path(f"{results_dir}/F1_scores/")
+f1_scores_dir.mkdir(parents=True, exist_ok=True)
+
+# load features data from indexes and features dataframe
+data_split_path = pathlib.Path("../1.split_data/indexes/data_split_indexes.tsv")
+data_split_indexes = pd.read_csv(data_split_path, sep="\t", index_col=0)
+features_dataframe_path = pathlib.Path("../0.download_data/data/training_data.csv.gz")
+features_dataframe = get_features_data(features_dataframe_path)
+
+
+# ### Evaluate Each Model on Each Dataset
+
+# In[3]:
+
+
+# directory to load the models from
+models_dir = pathlib.Path("../2.train_model/models/")
+
+# iterate through each model (final model, shuffled baseline model, etc)
+for model_path in models_dir.iterdir():
+    model = load(model_path)
+    model_name = model_path.name.replace("log_reg_","").replace(".joblib","")
+    
+    # iterate through label datasets (labels correspond to train, test, etc)
+    # with nested for loops, we test each model on each dataset(corresponding to a label)
+    for label in data_split_indexes["label"].unique():
+        print(f"Evaluating {model_name} on dataset {label}")
+        # load dataset (train, test, etc)
+        data = get_dataset(features_dataframe, data_split_indexes, label)
+        # path to save scores to
+        score_save_path = pathlib.Path(f"{f1_scores_dir}/{model_name}_{label}.tsv")
+        # find model scores on dataset
+        score = model_score(model, data)
+        
+        # save scores
+        # transpose data and reset index to make dataframe resemble tidy long format
+        score = score.T.reset_index()
+        # change columns to their respective names
+        score.columns = ["Phenotypic_Class", "F1_Score"]
+        score.to_csv(score_save_path, sep="\t")
```

Comment:
Rather than saving four separate files, consider only saving one but adding two columns: `data_split` (which will store train or test) and `shuffled` (which will store true or false).

This increases the size of the data slightly (higher redundancy than storing model conditions in file name), but it tends to be much easier to need to wrangle only one file rather than four. (plus, data are really small, so nbd)

Reply:
Done in https://github.com/WayScience/phenotypic_profiling_model/pull/9/commits/df1551dcbda4ee6ad9c77f2bc6d80fd4bca6c00d.